python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -407,6 +427,32 @@ def test_torch_tensor_nccl_nested_dynamic(ray_start_regular):
     compiled_dag.teardown()
 
 
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_within_same_actor(ray_start_regular, monkeypatch):
```

Comment:
This test seems to require GPU. Should we add

```python
    if not USE_GPU:
        pytest.skip("NCCL tests require GPUs")
```

Reply:
Another solution is to set the resource requirement for TrainWorker to CPU instead of GPU.

Reply:
I used CPU instead.